src/nw/ExchangePlatform/data/DTCCWarehouse.java

```diff
@@ -5,11 +5,11 @@
 
 public class DTCCWarehouse {
 
-    //<userID, MarketParticipantPortfolio>
-    HashMap<String, MarketParticipantPortfolio> portfolio = new HashMap<>();
+    //<ticker,HashMap<userID,certificate>>
+    public HashMap<String, HashMap<String,SecurityCertificate>> certificatesMap = new HashMap<>();
```

Comment:
Its probably better to refer to market participants with a market participant ID (long or int) rather than string

Reply:
sounds great. will work on it for the next pull request.

Reply:
sounds good. will work on and include the changes in the next pull request.